### PR TITLE
[tests] Fixed browser tests #280

### LINKED ISSUE #280: [CI] Driver is undefined in browser test
Sometimes jest `beforeAll` hook gave timeout in driver initialization.
We want to be specific while running tests.
`beforeAll` hook should wait till the driver is initialized and then the test must run.

The issue is mentioned here https://github.com/facebook/jest/issues/9527

We can use jest-circus as a test runner to solve this issue.

image

### PR DESCRIPTION
**Pull request by nemesisdesign**
Closes #280

### CODE REVIEW

File: browser-test/utils.js [hunk 1]
```diff
@@ -1,34 +1,31 @@
...
+
+  if (result.stdout) {
+    process.stdout.write(`${result.stdout}\n`);
+  }
+  if (result.stderr) {
+    process.stderr.write(`${result.stderr}\n`);
+  }
+  if (result.status !== 0) {
+    process.exit(result.status);
+  }
```
Comment (nemesisdesign) [inline browser-test/utils.js:21]: @codesankalp unfortunately I am no expert when it comes to nodeJS.
While testing I have noticed that the scripts keep running after the tests have finished and jest complains about it.
After debugging I found out that we could have used `spawnSync` which is a lot simpler and more suited for this kind of operation.

I also have seen there's `execSync` but would have probably meant to change a lot more the code so I didn't go for it.

File: browser-test/utils.js [hunk 2]
```diff
@@ -56,3 +53,5 @@ export const urls = {
   login: `http://0.0.0.0:8080/${orgSlug}/login`,
   status: `http://0.0.0.0:8080/${orgSlug}/status`,
 };
+
+jest.setTimeout(10000);
```
Comment (nemesisdesign) [inline browser-test/utils.js:59]: I increased the global jest time out here.

File: browser-test/utils.js [hunk 3]
```diff
@@ -1,34 +1,31 @@
...
 
-export const getDriver = async (Builder, options) => {
-  const driver = await new Builder()
+export const getDriver = async () => {
+  return new Builder()
     .forBrowser("firefox")
-    .setFirefoxOptions(options)
+    .setFirefoxOptions(new firefox.Options().headless())
     .build();
-  return driver;
```
Comment (nemesisdesign) [inline browser-test/utils.js:31]: @codesankalp I simplified this. No need to repeat the headless option and `Builder` in each test file.

File: .eslintrc.json [hunk 4]
```diff
@@ -34,6 +34,16 @@
     ],
     "react/jsx-props-no-spreading": 0,
     "react/forbid-prop-types": 0,
-    "react/sort-comp": [0]
+    "react/sort-comp": [0],
+    "import/no-extraneous-dependencies": [
```
Comment (nemesisdesign) [inline .eslintrc.json:38]: this is needed after moving the imports of selenium in `utils.js`

File: browser-test/utils.js (hunk 2 above)
Comment (codesankalp) [reply browser-test/utils.js:59]: Yes, it needs to be increased.

**Pull request merged**